optimizationProblem.m

```diff
@@ -98,6 +98,22 @@
                 obj.signs = zeros(obj.N - 1,1)+eps; % setting to zero flips out due to sqrt(0)=complex (??)
             end
             
+            if ~isempty(obj.motionCompensation.order)
+                
+                if isempty(obj.motionCompensation.linear)
+                    % Infer empty motionCompensation.linear from values of
+                    % motionCompensation.maxMagnitude
+                    obj.motionCompensation.linear = (obj.motionCompensation.maxMagnitude <= 0);
+                elseif length(obj.motionCompensation.linear) == 1
```

Comment:
Im not sure I see the point of this. It would seem that setting .linear to a scalar 1, forces linear optimization even when the magnitudes are > 1.

I would prefer that we always force the user to define the order and magnitude explicitly.

Reply:
From a technical standpoint it doesn't really matter, I guess it's more a question about what makes the most sense for the user and whether we want to force the "correct" behavior upon them. 

But I'm leaning towards accepting your proposal since linear vs. nonlinear can be considered an implementation detail that the user shouldn't be bothered with.